### DISCUSSION TITLE
Evaluate hiennguyenle/finn fork

### DISCUSSION THREAD

**Discussion by jothepro** [cross-language-cpp/djinni-generator#73] [Ideas]
*This idea has been discussed in issue #17 before. We decided that instead of the issue tracker we want to promote Github Discussions as a place to collect & discuss ideas that need more refinement or that may not be super concrete yet. That's why the discussion will continue here.*

@prsolucoes has mentioned this fork of djinni:
https://github.com/hiennguyenle/finn

At a first glance it seems very active & miles ahead in terms of features.

As a maintainer I want to evaluate the improvements. To list some awesome features that have been added to the generator:

- Deprecated Directive
- Extend Record
- a swift generator

Given that the advancements match with our idea of djinni, I'd like to get in contact with the maintainer if . Maybe we can collaborate in some way.

---- 

*Aspects that have been discussed in the issue to this point:*

- @paulo-coutinho used it before and it seems to be working (https://github.com/cross-language-cpp/djinni-generator/issues/17#issuecomment-756901419).
- @a4z has tried to contect the author of the fork and asked him if he is interested in contributing to the cross-language-cpp fork. (https://github.com/cross-language-cpp/djinni-generator/issues/17#issuecomment-882882220)

**Comment by a4z** [comment]
Since we haven't got any feedback on our request for pretty long time now, I think we can forget about that.

**Comment by jothepro** [comment]
@mutagene in response to your comment  https://github.com/cross-language-cpp/djinni-generator/issues/17#issuecomment-916241093 : 
> It's a pity this didn't work out. For me, I could certainly take advantage of generated json serialisers - something I'm currently doing a lot by hand with the djinni-generated types.

I think it shouldn't be a problem to copy the feature as long as you mention the original authors. It would have been cool to have the original maintainer on board, but if you have a need for the feature it would be cool if you'd contribute it here!

**Comment by mutagene** [comment]
I have made a PR which borrows heavily from hienguyenle fork and adds the subset of JSON functionality that interests me. I hope it interests others as well 😅 

https://github.com/cross-language-cpp/djinni-generator/pull/99
